extensions/amp-subscriptions/0.1/local-subscription-platform.js

```diff
@@ -14,235 +14,30 @@
-          // enhance this action, e.g. to provide save/link feature.
-          const platform = this.serviceAdapter_.selectPlatformForLogin();
-          this.serviceAdapter_.delegateActionToService(
-              action, platform.getServiceId());
-        } else {
-          this.executeAction(action);
-        }
-      } else if (serviceAttr) {
-        this.serviceAdapter_.delegateActionToService(action, serviceAttr);
-      }
-    }
-  }
-
-  /** @override */
-  activate(entitlement) {
-    const renderState = entitlement.json();
-    this.urlBuilder_.setAuthResponse(renderState);
-    this.actions_.build().then(() => {
-      this.renderer_.render(renderState);
-    });
-  }
-
-  /** @override */
-  reset() {
-    this.renderer_.reset();
-  }
-
-  /** @override */
-  executeAction(action) {
-    const actionExecution = this.actions_.execute(action);
-    return actionExecution.then(result => {
-      if (result) {
-        this.serviceAdapter_.resetPlatforms();
-      }
-      return !!result;
-    });
-  }
-
-  /** @override */
-  isPrerenderSafe() {
-    // Local platform can never be allowed to prerender in a viewer
-    return false;
-  }
-
-  /** @override */
-  getEntitlements() {
-    return this.urlBuilder_.buildUrl(this.authorizationUrl_,
-        /* useAuthData */ false)
-        .then(fetchUrl => {
-          const encryptedDocumentKey =
-              this.serviceAdapter_.getEncryptedDocumentKey('local');
-          if (encryptedDocumentKey) {
-            //TODO(chenshay): if crypt, switch to 'post'
-            fetchUrl = addParamToUrl(fetchUrl, 'crypt', encryptedDocumentKey);
-          }
-          this.xhr_.fetchJson(fetchUrl, {credentials: 'include'})
-              .then(res => res.json())
-              .then(resJson => {
-                return Entitlement.parseFromJson(resJson);
-              });
-        });
-  }
-
-  /** @override */
-  isPingbackEnabled() {
-    return !!this.pingbackUrl_;
-  }
-
-  /** @override */
-  pingback(selectedEntitlement) {
-    if (!this.isPingbackEnabled) {
-      return;
-    }
-    const pingbackUrl = /** @type {string} */ (devAssert(this.pingbackUrl_,
-        'pingbackUrl is null'));
-
-    const promise = this.urlBuilder_.buildUrl(pingbackUrl,
-        /* useAuthData */ true);
-    return promise.then(url => {
-      // Content should be 'text/plain' to avoid CORS preflight.
-      return this.xhr_.sendSignal(url, {
-        method: 'POST',
-        credentials: 'include',
-        headers: dict({
-          'Content-Type': 'text/plain',
-        }),
-        body: JSON.stringify(selectedEntitlement.jsonForPingback()),
-      });
-    });
-  }
-
-  /** @override */
-  getSupportedScoreFactor(unusedFactor) {
-    return 0;
-  }
-
-  /** @override */
-  getBaseScore() {
-    return this.serviceConfig_['baseScore'] || 0;
-  }
-
-  /** @override */
-  decorateUI(unusedNode, unusedAction, unusedOptions) {}
+export function LocalSubscriptionPlatformFactory(
```

Comment:
nit: since it's a function, let's start with a lower-case "l".

Reply:
fixed